src/configs/astro.ts

```diff
@@ -64,8 +78,38 @@ export async function astro(
             }
           : {},
 
+        ...formatter
+          ? DISABLE_STYLISTIC_RULES
+          : {},
+
         ...overrides,
       },
     },
+    {
+      // Define the configuration for `<script>` tag when using `client-side-ts` processor.
+      files: ['**/*.astro/*.ts', '*.astro/*.ts'],
+      languageOptions: {
+        globals: {
+          ...globals.browser,
+        },
+        parser: parserTs,
+        parserOptions: {
+          project: null,
+        },
+        sourceType: 'module',
+      },
+      // Script in `<script>` is assigned a virtual file name with the `.ts` extension.
+      name: 'antfu/astro/base/typescript',
+      rules: {
+        'prettier/prettier': 'off',
+
+        ...formatter
+          ? DISABLE_STYLISTIC_RULES
+          : {},
+
+        // Type aware rules breaks the astro plugin
+        ...(pluginTs?.configs?.['disable-type-checked']?.rules ?? {}),
```

Comment:
Is this permanent or is it a temporary bug of the Astro plugin?

If astro can't get type-checked, I would think it's better to exclude them in the typecheck config, instead of disabling them here (which could be different rule sets)

Reply:
I move the typeware rules to the typescript config and also I notice that I can do the same for the formatter. If the formatter is enable, the formatter can be in charge of disable the stylistic rules for astro (this new eslint flat config is really good).